tests/metrics/run.sh

```diff
@@ -0,0 +1,125 @@
+	# check two worker's secondsBehindMaster > 0
+	check_secondsBehindMaster 0 2
+	echo "check ddl done!"
+
+	# restart dm worker
+	kill_dm_worker
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/BlockSyncerUpdateLag=return(\"insert,2\");github.com/pingcap/dm/syncer/changeTickerInterval=return(10)"
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+
+	run_sql_source1 "truncate table metrics.t1;"                                        # make skip job
+	run_sql_file $cur/data/db1.increment.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1 # make dml job
+	run_sql_file $cur/data/db2.increment.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2 # make dml job
+
+	# test dml lag metric >= 2 beacuse we inject updateReplicationLag(insert) to sleep(2)
+	# although skip lag is 0 (locally), but we use that lag of all dml/skip lag, so lag still >= 2
+	check_metric $WORKER1_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 1 999
+	check_metric $WORKER2_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 1 999
+	check_secondsBehindMaster 1 2
+	echo "check dml/skip done!"
+
+	# restart dm worker
+	kill_dm_worker
+	export GO_FAILPOINTS=''
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+	# check the dmctl query-status no new dml, lag should be set to 0
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status test" \
+		"\"secondsBehindMaster\": \"0\"" 2
+	echo "check zero job done!"
+
+	kill_dm_worker
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/changeTickerInterval=return(5)"
+	# First set the ticker interval to 5s -> expect the execSQL interval to be greater than 5s
+	# At 5s, the first no job log will appear in the log
+	# At 6s, the ticker has already waited 1s and the ticker goes to 1/5th of the way
+	# At 6s, a dml job is added to jobchan and the ticker is reset
+	# At 11s the ticker  wirte the log of the second nojob
+	# Check that the interval between the two ticker logs is > 5s
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	echo "sleep 6s"
+	sleep 6
+	echo "make a dml job"
+	run_sql_source1 "use metrics;insert into t1 (id, name, ts) values (1004, 'zmj4', '2022-05-11 12:01:05')"
+	echo "sleep 5s"
+	sleep 5
```

Comment:
maybe sleep slightly longer than 5 seconds, or add some retry, to avoid a unstable test.

Reply:
current test metrics task Less than 1 minutes，and  `5s` interval basically guarantees that the test will pass

* 1~3 second is  unstable
* 4 second It looks unlucky 😬

Reply:
I'm afraid the worker runs slower so it takes **more than** 5 seconds to log, so we might sleep 6 seconds for example